pkg/apis/core/types.go

```diff
@@ -2277,6 +2277,10 @@ type Container struct {
 	// +featureGate=InPlacePodVerticalScaling
 	// +optional
 	ResizePolicy []ContainerResizePolicy
+	// Restart policy for the container.
```

Comment:
Shouldn't this describe what it is for, the valid options, and the limitations on its use?

I know we talked about using different values vs same values - does it need to be the same Go type, or should we define a new Go type with comments that reflect the per-container use?

Reply:
+1. The v1 type (https://github.com/kubernetes/kubernetes/pull/116429#discussion_r1131457269) has slightly more documentation. Probably worth keeping these in sync as well.

Reply:
> does it need to be the same Go type, or should we define a new Go type with comments that reflect the per-container use?

Maybe we can make a new `ContainerRestartPolicy` type.

@SergeyKanzhelev 
What do you think?

Reply:
> Shouldn't this describe what it is for, the valid options, and the limitations on its use?

added more description.

Reply:
> Maybe we can make a new `ContainerRestartPolicy` type.

yes, I think it should be a separate type

Reply:
added a new type and its description.